ksonnet-gen/ksonnet/renderer.go

```diff
@@ -0,0 +1,362 @@
+package ksonnet
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	nm "github.com/ksonnet/ksonnet-lib/ksonnet-gen/nodemaker"
+	"github.com/pkg/errors"
+)
+
+// renderer is an item that can be rendered.
+type renderer interface {
+	Render(parent *nm.Object) error
```

Comment:
Noticed that none of the implementations return errors. Maybe better to remove it from the signature so there's one less thing to handle. I'm fine keeping it too though if you anticipate future implementations to potentially need to return an error.

Reply:
This is a forward looking thing. Also, LiteralFieldRenderer#Render actually does return an error. I did this because this a library and wanted some flexibility for the future.